### ISSUE TITLE
TypingError when iterating over homogeneous tuples

### ISSUE THREAD

**Issue by jcrist** [numba/numba#1504]
```python
In [30]: @nb.jit(nopython=True)
def foo(data):
    return data[0] + data[1]
   ....:

In [31]: @nb.jit(nopython=True)
def bar(x, y):
    total = 0
    for row in zip(x, y):
        total += foo(row)
    return total
   ....:

In [32]: x = y = np.arange(3)

In [33]: foo((x[0], y[0]))    # Works fine
Out[33]: 0

In [34]: bar(x, y)    # Typing error
---------------------------------------------------------------------------
TypingError                               Traceback (most recent call last)
...
Failed at nopython (nopython frontend)
Invalid usage of getitem with parameters ((int64, int64), int64)
 * parameterized
```

Works to index into a homogeneous tuple of `(int64, int64)`, but fails if iterating over a series of them. Numba version 0.21.0.

**Comment by sklam** [comment]
the error is due to `row` being typed to heterogenous tuple

**Comment by jcrist** [comment]
Fixed by #1505. Closing.
